src/main/kotlin/com/arsiu/eduhub/service/CourseService.kt

```diff
@@ -14,41 +13,28 @@ import org.springframework.stereotype.Service
 @Service
 class CourseService @Autowired constructor(
     private val courseRepository: CourseRepository,
-    private val userService: UserServiceInterface,
-    @Lazy val chapterService: ChapterServiceInterface
+    @Lazy private val userService: UserServiceInterface
 ) : CourseServiceInterface {
 
     override fun findAll(): List<Course> = courseRepository.findAll().toList()
 
-    override fun findById(id: Long): Course =
+    override fun findById(id: String): Course =
         courseRepository.findById(id).orElseThrow { NotFoundException("Course with ID $id not found") }
 
     @NotifyTrigger("New Course Available ")
     override fun create(entity: Course): Course {
         entity.owner = userService.findById(entity.owner.id)
-
-        val chapters = entity.chapters.toList()
-
-        entity.chapters.clear()
-
-        val createdCourse = courseRepository.save(entity)
-
-        val updatedChapters = chapters.map { chapter ->
-            chapter.course = createdCourse
-            chapterService.create(chapter)
-        }
-
-        createdCourse.chapters.addAll(updatedChapters)
-
-        return courseRepository.save(createdCourse)
+        return courseRepository.createCascade(entity)
     }
 
-    override fun update(id: Long, entity: Course) : Course {
-        userService.findById(entity.owner.id)
+    override fun update(id: String, entity: Course): Course {
         entity.id = findById(id).id
+        delete(id)
```

Comment:
maybe you need to create some custom update method, rather than delete and create?

Reply:
done, reworked implementation from full replacement to particular replacement